sdk/openai/openai/test/public/assisstants.spec.ts

```diff
@@ -0,0 +1,237 @@
+
+            const metadataValue = "bar";
+            const messageOptions = {
+              metadata: { foo: metadataValue },
+            };
+            const messageResponse = await client.beta.threads.messages.create(threadResponse.id, {
+              role,
+              content,
+              ...messageOptions,
+            });
+
+            let messageContent = messageResponse.content[0];
+            assert.isNotNull(messageResponse.id);
+            assert.equal(messageResponse.role, role);
+            if (messageContent.type === "text") {
+              assert.equal(messageContent.text.value, content);
+            }
+            assert.equal((messageResponse.metadata as any).foo, metadataValue);
+            const getMessageResponse = await client.beta.threads.messages.retrieve(
+              threadResponse.id,
+              messageResponse.id || "",
+            );
+            messageContent = getMessageResponse.content[0];
+            assert.equal(messageResponse.id, getMessageResponse.id);
+            assert.equal(getMessageResponse.role, role);
+            if (messageContent.type === "text") {
+              assert.equal(messageContent.text.value, content);
+            }
+            assert.equal((getMessageResponse.metadata as any).foo, metadataValue);
+
+            const newMetadataValue = "other value";
+            messageOptions.metadata.foo = newMetadataValue;
+
+            const updateMessageResponse = await client.beta.threads.messages.update(
+              threadResponse.id,
+              messageResponse.id || "",
+              messageOptions,
+            );
+            assert.equal(messageResponse.id, updateMessageResponse.id);
+            assert.equal((updateMessageResponse.metadata as any).foo, newMetadataValue);
+
+            const listLength = 1;
+            const oneMessageList = await client.beta.threads.messages.list(threadResponse.id, {
+              limit: listLength,
+            });
+            assert.equal(oneMessageList.data.length, listLength);
+            const firstID = (oneMessageList as any).body.first_id;
+            const lastID = (oneMessageList as any).body.last_id;
+            assert.equal(firstID, lastID);
+            assert.equal(oneMessageList.data[0].id, firstID);
+          });
+
+          it("uploads, gets, and lists a file", async function () {
+            /*
+          // move to node only, currently failing in browser
+          const filename = "sample_file_for_upload.txt";
+          const text = "The word 'apple' uses the code 442345, while the word 'banana' uses the code 673457.";
+          const uint8 = Uint8Array.from(text.split("").map(x => x.charCodeAt(0)));
+          const uploadedFile = await client.uploadFile(uint8, "assistants", { filename });
+          assert.isNotNull(uploadedFile.id);
+          assert.equal(uploadedFile.filename, filename);
+          assert.equal(uploadedFile.bytes, uint8.length);
+          */
```

Comment:
why is this commented out? We could move this test to its own module under a node folder.

Reply:
I copied this from Glenn's assistant test suite so I haven't double check this test, I can create the test into a new file.